clcache.py

```diff
@@ -737,6 +737,21 @@ def expandCommandLine(cmdline):
 class CommandLineAnalyzer(object):
 
     @staticmethod
+    def _preprocessToStdout(options):
+        assert ('P' in options or 'E' in options or 'EP' in options), "must be in preprocessor mode"
+
+        # Note: For MSVS 2013 and 2015 there is a documentation bug in
+        # https://msdn.microsoft.com/en-us/library/becb7sys.aspx stating that
+        # "To send the preprocessed output to stdout, with #line directives,
+        # use /P and /EP together."
+        # The actual compiler behavior is that both /P and /P /EP go to file.
+        # /P is with #line annotations and /P /EP is without.
+        if 'P' in options:
+            return False
+        else:
+            return True
```

Comment:
My rule of thumb is to place assertions whenever a failing expression is a programming error.

This method must never be called when not in preprocessing because the entire logic only woks there. So instead of pushing the error down the road where it gets harder to debug, we should fail fast.

I am fine with rewriting the implementation in one line but I think the assertion adds value.